Dockerfile

```diff
@@ -19,22 +19,39 @@ FROM --platform=$TARGETPLATFORM debian:stable-slim AS buildbase
 # built-in arg set by `docker build --platform linux/$TARGETARCH ...`
 ARG TARGETARCH
 ARG CONDA_VERSION=latest
+ARG conda_docker_default_channel=defaults
 
 WORKDIR /tmp
 
 RUN apt-get update && apt-get install -y wget
 
-RUN if [ "${TARGETARCH}" = "amd64" ]; then \
-        MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-x86_64.sh"; \
-    elif [ "${TARGETARCH}" = "s390x" ]; then \
-        MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-s390x.sh"; \
-    elif [ "${TARGETARCH}" = "arm64" ]; then \
-        MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-aarch64.sh"; \
-    elif [ "${TARGETARCH}" = "ppc64le" ]; then \
-        MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-ppc64le.sh"; \
+RUN if [ "${conda_docker_default_channel}" = "defaults" ]; then \
+        if [ "${TARGETARCH}" = "amd64" ]; then \
+            MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-x86_64.sh"; \
+        elif [ "${TARGETARCH}" = "s390x" ]; then \
+            MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-s390x.sh"; \
+        elif [ "${TARGETARCH}" = "arm64" ]; then \
+            MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-aarch64.sh"; \
+        elif [ "${TARGETARCH}" = "ppc64le" ]; then \
+            MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-ppc64le.sh"; \
+        else \
+            echo "Not supported source channel & target architecture: ${conda_docker_default_channel} & ${TARGETARCH}"; \
+            exit 1; \
+        fi; \
+    elif [ "${conda_docker_default_channel}" = "conda-forge" ]; then \
+        if [ "${TARGETARCH}" = "amd64" ]; then \
+            MINICONDA_URL="https://github.com/conda-forge/miniforge/releases/${CONDA_VERSION}/download/Miniforge3-Linux-x86_64.sh"; \
+        elif [ "${TARGETARCH}" = "arm64" ]; then \
+            MINICONDA_URL="https://github.com/conda-forge/miniforge/releases/${CONDA_VERSION}/download/Miniforge3-Linux-aarch64.sh"; \
+        elif [ "${TARGETARCH}" = "ppc64le" ]; then \
+            MINICONDA_URL="https://github.com/conda-forge/miniforge/releases/${CONDA_VERSION}/download/Miniforge3-Linux-ppc64le.sh"; \
+        else \
+            echo "Not supported source channel & target architecture: ${conda_docker_default_channel} & ${TARGETARCH}"; \
+            exit 1; \
+        fi; \
     else \
-        echo "Not supported target architecture: ${TARGETARCH}"; \
-        exit 1; \
+        echo "conda_docker_default_channel value ${conda_docker_default_channel} not supported"; \
+        exit 1; \ 
```

Comment:
@dbast probably has opinions on this, but could we try to move some of this into a script so it's not as painful to look at?